clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1016,6 +1000,11 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
     if (!IsThinLTOPostLink) {
       addSanitizers(TargetTriple, CodeGenOpts, LangOpts, PB);
       addKCFIPass(TargetTriple, LangOpts, PB);
+      PB.registerPipelineStartEPCallback(
```

Comment:
we should be consistent and make both entry exit instrumenters part of the default pipelines, not a clang-specific add-on

this would probably be somewhere in `buildModuleSimplificationPipeline`, checking `Phase` (and make sure -O0 works by modifying `buildO0DefaultPipeline`)

Reply:
Done.